### PR TITLE
SharePanel: Add icons to panel options Share and More

### LINKED ISSUE #92612: SharePanel: Add icons to panel options Share and More
**Why is this needed**:
Add icons to Share and More options of Panels
Here is the [figma](https://www.figma.com/design/AjvrAorYpnfSoMpU5kwf39/Adoption-of-Sharing?node-id=4876-1790&m=dev#921770184) 

**What would you like to be added**:
Add icons to Share and More options of Panels

**Who is this feature for?**

Users

### PR DESCRIPTION
**Pull request by lucychen-grafana**
**What is this feature?**
Add icons to Share and More options of Panels
Here is the [figma](https://www.figma.com/design/AjvrAorYpnfSoMpU5kwf39/Adoption-of-Sharing?node-id=4876-1790&m=dev#921770184)

**Why do we need this feature?**
Add icons to Share and More options of Panels for better user experience

**Who is this feature for?**
Customers

**Which issue(s) does this PR fix?**:

Fixes #https://github.com/grafana/grafana/issues/92612

**Special notes for your reviewer:**

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/contribute/release-notes/#how-to-determine-if-content-belongs-in-whats-new), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/contribute/release-notes/) doc.

### CODE REVIEW

File: packages/grafana-data/src/types/icon.ts [hunk 1]
```diff
@@ -166,9 +166,12 @@ export const availableIconsIndex = {
   kubernetes: true,
   'layer-group': true,
   'layers-alt': true,
+  'legend-show': true,
+  'legend-hide': true,
   'library-panel': true,
   'line-alt': true,
   link: true,
+  'link-broken 1': true,
```
Comment (tskarhed) [inline packages/grafana-data/src/types/icon.ts:174]:
```suggestion
  'link-broken-1': true,
```
Is this possible for consistency? Or is the 1 needed at all?

Comment (juanicabanas) [reply packages/grafana-data/src/types/icon.ts:174]: is there any reason why it is 1? What about `link-broken` ?

File: packages/grafana-data/src/types/icon.ts [hunk 2]
```diff
@@ -166,9 +166,12 @@ export const availableIconsIndex = {
   kubernetes: true,
   'layer-group': true,
   'layers-alt': true,
+  'legend-show': true,
+  'legend-hide': true,
```
Comment (ashharrison90) [inline packages/grafana-data/src/types/icon.ts:170]: this is really minor, but if this list is alphabetical then `legend-hide` should be before `legend-show`

File: public/img/icons/unicons/legend-show.svg [hunk 3]
```diff
@@ -0,0 +1,7 @@
+<svg width="48" height="48" viewBox="0 0 48 48" xmlns="http://www.w3.org/2000/svg">
```
Comment (ashharrison90) [inline public/img/icons/unicons/legend-show.svg:1]: please see the guidelines [here](https://github.com/grafana/grafana/blob/main/public/img/icons/README.md) for how to properly format these icons. in particular, i think you'll need to remove the explicit `width` and `height` attributes

**Review by ashharrison90 — COMMENTED** [review]
couple of nits. agree with the others about the name (and filename) should just be `link-broken`

@Clarity-89 how does this work with the new icon package? 🤔 have to admit i can't really remember what the state of that is 😅

**Comment by Clarity-89** [comment]
> @Clarity-89 how does this work with the new icon package? 🤔 have to admit i can't really remember what the state of that is 😅

Yeah, it's not ideal for the new icons, as those will be components. Maybe we'll be able to use component names there, but since we don't have an ETA on when the icons package will be published, I don't think we'd optimise for that currently.

File: packages/grafana-data/src/types/icon.ts (hunk 1 above)
Comment (lucychen-grafana) [reply packages/grafana-data/src/types/icon.ts:174]: There's already a [link-broken icon ](https://github.com/grafana/grafana/blob/main/public/img/icons/unicons/link-broken.svg). It looks the same so I'll just use this instead

**Review by juanicabanas — APPROVED** [review]
LGTM

**Pull request merged**